src/scrot.c

```diff
@@ -400,38 +400,54 @@ Window scrotGetWindow(Display *display, Window window, int x, int y)
 void scrotGrabMousePointer(const Imlib_Image image, const int xOffset,
     const int yOffset)
 {
-    XFixesCursorImage *xcim = XFixesGetCursorImage(disp);
-
-    if (!xcim) {
-        warnx("Failed to get mouse cursor image.");
-        return;
+    XFixesCursorImage *xcim = NULL;
+    char *errmsg = NULL;
+    unsigned short width;
+    unsigned short height;
+    int x;
+    int y;
+    size_t pixcnt;
+    uint32_t *pixels = NULL;
+    Imlib_Image imcursor;
+
+    /* Convert an X cursor into the format imlib wants. */
+    if ((xcim = XFixesGetCursorImage(disp)) == NULL) {
+        errmsg = "Can't get the cursor from X";
```

Comment:
Also this entire `errmsg` seems completely unnecessary. Why not just do the following:

```c
if (...) {
        warn("msg");
        goto end;
}
```